simulation/halsim_gui/src/main/native/include/HALSimGui.h

```diff
@@ -10,6 +10,8 @@
 
 #include <memory>
```

Comment:
This should also include <functional> due to adding std::function (include-what-you-use).